### PR TITLE
Issue 2906: Document fix Pravega overview

### LINKED ISSUE #2906: Document Review for Pravega overview
**Problem description**
1.Fixing formatting errors
2. Checking broken links
3. Editing content

**Problem location**
/documentation/index.md

### PR DESCRIPTION
**Pull request by Aparnarr**
**Change log description**  
1. Fixed formatting errors
2. Edited content

**Purpose of the change**  
Fixes #2906

### CODE REVIEW

File: documentation/src/docs/segment-store-service.md [hunk 1]
```diff
@@ -9,236 +9,246 @@ You may obtain a copy of the License at
...
+The Pravega Segment Store Service is a subsystem that lies at the heart of the entire Pravega deployment. It is the main access point for managing Stream Segments, providing the ability to _create_, _delete_ and _modify/access_ their contents. The Pravega Client communicates with the Pravega Stream Controller to figure out which Segments need to be used (for a Stream), and both the Stream Controller and the Client deal with the Segment Store Service to actually operate on them.
 
-The basic idea behind the Segment Store Service is that it buffers incoming data in a very fast and durable append-only medium (Tier 1), and syncs it to a high-throughput (but not necessarily low latency) system (Tier 2) in the background, while aggregating multiple (smaller) operations to a Segment into a fewer (but larger) ones.
+The basic idea behind the Segment Store Service is that, it buffers the incoming data in a very fast and durable append-only medium (Tier 1), and syncs it to a high-throughput (but not necessarily low latency) system (Tier 2) in the background, while aggregating multiple (smaller) operations to a Segment into a fewer (but larger) ones.
```
Comment (andreipaduroiu) [inline documentation/src/docs/segment-store-service.md]: Why the comma here? It breaks the flow of the sentence.

File: documentation/src/docs/segment-store-service.md [hunk 2]
```diff
@@ -9,236 +9,246 @@ You may obtain a copy of the License at
...
 - Stream Segments that are unlimited in length, with append-only semantics, yet supporting arbitrary-offset reads.
-- No throughput degradation when performing small appends, regardless of the performance of the underlying Tier-2 storage system.
+- No throughput degradation when performing small appends, regardless of the performance of the underlying Tier 2 storage system.
 - Multiple concurrent writers to the same Segment.
  - Order is guaranteed within the context of a single writer, but appends from multiple concurrent writers will be added in the order in which they were received (appends are atomic without interleaving their contents).
 - Writing to and reading from a Segment concurrently with relatively low latency between writing and reading.
 
 # Terminology
-Throughout the rest of this document, we will use the following terminology:
+The follwoing terminologies are used throughout the document:
```
Comment (andreipaduroiu) [inline documentation/src/docs/segment-store-service.md]: typo

File: documentation/src/docs/index.md [hunk 3]
```diff
@@ -15,22 +15,21 @@ Read [Pravega Concepts](pravega-concepts.md) for more details.
...
 
--   Unlimited Retention - Ingest, process and retain data in streams forever. Use same paradigm to access both real-time and historical events stored in Pravega.
+-   Unlimited Retention:- Pravega performs ingestion, processing and retains data in streams forever. Pravega uses the same paradigm to access both real-time and historical events stored in it.
 
--   Storage Efficiency - Use Pravega to build pipelines of data processing, combining batch, real-time and other applications without duplicating data for every step of the pipeline.
+-   Storage Efficiency:-  Pravega is used to build pipelines of data processing. It is used to combine batch, real-time and other applications without duplicating data for every step of the pipeline.
 
--   Durability - Don't compromise between performance, durability and consistency.
-    Pravega persists and protects data before the write operation is acknowledged to the client.
+-   Durability:- Pravega do not compromise between performance, durability and consistency. Pravega persists and protects data before the write operation is acknowledged to the client.
```
Comment (RaulGracia) [inline documentation/src/docs/index.md]: Typo: `Pravega does not`. Moreover, I don't like the first sentence, because it is probably not true. There is an inherent trade-off relating performance and durability/consistency. That is, if you want durability and consistency guarantees, this will probably entail a set of tasks to be done that will necessary make the system slower compared to another one that does not provide such guarantees. But not only this, the second sentence says `before the write operation is acknowledged to the client`; I think that this is not true neither. If a writer writes an events and gets disconnected before getting the acknowledgement, there is not guarantee that the server has correctly stored this data event. Conversely, once an event is confirmed to the writer, we are sure that it has been correctly persisted. 

I would change this bullet as follows:
`Durability: Pravega persists and protects data events once the write operation is acknowledged to the client.`

File: documentation/src/docs/segment-store-service.md [hunk 4]
```diff
@@ -9,236 +9,246 @@ You may obtain a copy of the License at
...
 
-More details about each component described above can be found in the **Components** section (below).
+More details about each component described above can be found in the [Components](#Components) section.
 
 ## System Diagram
 
 ![System Diagram](img/Segment.Store.Components.png)
 
-In this image, we show the major components of the Segment Store (for simplicity, only one Segment Container is depicted). All Container components and major links between them (how they interact with each other) are shown. The _Container Metadata_ component is not shown, because every other component communicates with it in one form or another, and adding it would only clutter the diagram.
+In the above diagram, the major components of the Segment Store is shown and for simplicity, only one Segment Container is depicted. All Container components and major links between them (how they interact with each other) are shown. The _Container Metadata_ component is not shown, because every other component communicates with it in one form or another, and adding it would only clutter the diagram.
```
Comment (andreipaduroiu) [inline documentation/src/docs/segment-store-service.md]: are shown

Comment (andreipaduroiu) [reply documentation/src/docs/segment-store-service.md]: comma before "for simplicity"

File: documentation/src/docs/segment-store-service.md [hunk 5]
```diff
@@ -9,236 +9,246 @@ You may obtain a copy of the License at
...
-- _Cache_: Used to store data for appends that exist in Tier-1 only (not yet in Tier-2), as well as blocks of data that support reads.
-- _Storage Writer_: Processes the durable log operations and applies them to Tier-2 storage (in the order in which they were received). This component is also the one that coalesces multiple operations together, for better back-end throughput.
+- _Segment Container Metadata_: A collection of Segment-specific metadata that describe the current state of each Segment (how much data in Tier 2, how much in Tier 1, whether it is sealed, etc.), as well as other miscellaneous info about each Container.
+- _Durable Log_: The Container writes every operation it receives to this log and acks back only when the log says it has been accepted and durably persisted.
+- _Read Index_: An in-memory index of where data can be read from. The Container delegates all read requests to it, and it is responsible for fetching the data from wherever it is currently located (Cache, Tier 1 Storage or Tier2 Storage).
```
Comment (andreipaduroiu) [inline documentation/src/docs/segment-store-service.md]: "Tier 2" (to  be consistent with the rest of your changes)

File: documentation/src/docs/segment-store-service.md [hunk 6]
```diff
@@ -9,236 +9,246 @@ You may obtain a copy of the License at
...
+- _Metadata Operations_ are auxiliary operations that indicate a change to the Container Metadata. They can be the result of an external operation (we received an request for a Segment we never knew about before, so we must assign a unique Id to it) or to snapshot the entire Metadata (which helps with recovery and cleaning up Tier 1 Storage). The purpose of the _Metadata Operations_ is to reduce the amount of time needed for failover recovery (when needed).
+    - `StreamSegmentMapOperation`: Maps an Id to a Segment Name.
+    - `TransactionMapOperation`: Maps an Id to a Transaction and to its Parent Segment.
     - `UpdateAttributesOperation`: Updates any attributes on a Segment.
-    - `MetadataCheckpoint` includes an entire snapshot of the Metadata. This can be useful when during recovery - this contains all metadata up to this point, which is a sufficient base for all operations after it.
+    - `MetadataCheckpoint`: Includes an entire snapshot of the Metadata. This can be useful during recovery.This contains all metadata up to this point, which is a sufficient base for all operations after it.
```
Comment (andreipaduroiu) [inline documentation/src/docs/segment-store-service.md]: space between period and "This"

File: documentation/src/docs/index.md [hunk 7]
```diff
@@ -15,22 +15,21 @@ Read [Pravega Concepts](pravega-concepts.md) for more details.
...
     
--   Transaction Support - A developer uses a Pravega Transaction to ensure that a set of events are written to a stream atomically.
+-   Transaction Support:-  Pravega Transaction is used by the developer to ensure that a set of events are written to a stream atomically.
```
Comment (RaulGracia) [inline documentation/src/docs/index.md]: I don't like the use of the passive voice in such short, direct sentences. I would change this bullet as follows:
`Transaction Support: A Pravega Transaction ensures that a set of events are written to a stream atomically. This is a key feature for distributed streaming applications requiring exactly-once guarantees on their output.`

File: test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java [hunk 8]
```diff
@@ -51,6 +51,7 @@
     private static final int NUM_EVENTS = 100;
     private static final int TEST_ITERATIONS = 3;
     @Rule
+
```
Comment (RaulGracia) [inline test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java]: This change is unrelated with this PR, please remove.

**Comment by Aparnarr** [comment]
Closing this PR in favour of #2929
